herddb-core/src/main/java/herddb/storage/DataStorageManager.java

```diff
@@ -64,6 +64,10 @@
     public abstract List<Record> readPage(String tableSpace, String uuid, Long pageId)
         throws DataStorageManagerException, DataPageDoesNotExistException;
 
+    public abstract void ensureIndexDirectory(String tableSpace, String indexName) throws DataStorageManagerException;
```

Comment:
DataStorageManager is some more "generic"... shouldn't be better a initIndex, bootIndex or prepareIndex?

indexName is unfortunately not really... true.. some index use uuid other a composite name. We can name it as uuid the is just little less misleading (because who use the uuid... has a name too... BRIN indexes for example)

Reply:
renamed to "bootIndex"